pdb2sql/StructureSimilarity.py

```diff
@@ -96,48 +100,29 @@ def compute_lrmsd_fast(self,lzone=None,method='svd',check=True):
         if check:
 
             # Note:
-            # 1. read_data_zone returns in_zone and not_in_zone
-            # which means the in_zone only defines the zone for fitting
-            # but not for rmsd calculation.
-            # 2. the decoy and ref pdb must have consitent residue
+            # 1. get_data_zone_backbone returns in_zone and not_in_zone
+            #  here the in_zone defines the zone for fitting,
+            #  and not_in_zone defines the zone for rms calculation.
+            # 2. the decoy and ref pdb must have consistent residue
             # numbering, otherwise e.g. shifted numbering can also give
             # results which is totally wrong, because the code here does
             # not do sequence alignment.
 
-            data_decoy_long, data_decoy_short  = self.read_data_zone(self.decoy,resData,return_not_in_zone=True)
-            data_ref_long,   data_ref_short    = self.read_data_zone(self.ref,resData,return_not_in_zone=True)
-
-            atom_decoy_long = [ data[:3] for data in data_decoy_long ]
-            atom_ref_long   = [ data[:3] for data in data_ref_long ]
-
-            xyz_decoy_long, xyz_ref_long = [],[]
-            for ind_decoy, at in enumerate(atom_decoy_long):
-                try:
-                    ind_ref = atom_ref_long.index(at)
-                    xyz_decoy_long.append(data_decoy_long[ind_decoy][3:])
-                    xyz_ref_long.append(data_ref_long[ind_ref][3:])
-                except ValueError:
-                    warnings.warn(
-                        f'Decoy atom {at} not found in reference pdb')
-
-            atom_decoy_short = [data[:3] for data in data_decoy_short]
-            atom_ref_short   = [data[:3] for data in data_ref_short]
-
-            xyz_decoy_short, xyz_ref_short = [],[]
-            for ind_decoy, at in enumerate(atom_decoy_short):
-                try:
-                    ind_ref = atom_ref_short.index(at)
-                    xyz_decoy_short.append(data_decoy_short[ind_decoy][3:])
-                    xyz_ref_short.append(data_ref_short[ind_ref][3:])
-                except ValueError:
-                    warnings.warn(
-                        f'Decoy atom {at} not found in reference pdb')
+            data_decoy_long, data_decoy_short  = self.get_data_zone_backbone(self.decoy,resData,return_not_in_zone=True)
+            data_ref_long,   data_ref_short    = self.get_data_zone_backbone(self.ref,resData,return_not_in_zone=True)
+
+            atom_long = data_ref_long.intersection(data_decoy_long)
```

Comment:
this intersection stuff does all the work ?

Reply:
It gets the common residues or atoms and ignore missing ones. It's  actually checking if the sequences are consistent or not, but it cannot fix it if not consistent(even the former code cannot do this).